src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -7,8 +7,32 @@
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
-    @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationException(
+                    "Login cannot be null");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException(
+                    "User with this login already exists");
+        }
+
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationException(
+                    "Login must be at least 6 characters long");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new RegistrationException(
+                    "Password must be at least 6 characters long");
+        }
+        if (user.getAge() < 18) {
+            throw new RegistrationException(
+                    "User must be at least 18 years old");
+        }
+
+        storageDao.add(user);
+
+        return user;
```

Comment:
The `add` method in `StorageDao` returns the added User. You should return the result of the `add` method directly, instead of returning the input parameter `user`. This way, you ensure that the returned User is the one that has been actually stored.